internal/service/handlers/verify_passport.go

```diff
@@ -1,232 +1,100 @@
+	zkptypes "github.com/iden3/go-rapidsnark/types"
+	"github.com/rarimo/decentralized-auth-svc/pkg/auth"
 	"github.com/rarimo/rarime-points-svc/internal/data"
 	"github.com/rarimo/rarime-points-svc/internal/data/evtypes"
 	"github.com/rarimo/rarime-points-svc/internal/service/requests"
-	"github.com/rarimo/rarime-points-svc/pkg/connector"
+	zk "github.com/rarimo/zkverifier-kit"
 	"gitlab.com/distributed_lab/ape"
 	"gitlab.com/distributed_lab/ape/problems"
+	"gitlab.com/distributed_lab/logan/v3/errors"
 )
 
 func VerifyPassport(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusNoContent)
-	return
-
-	// TODO: New logic. User go to public endpoint with proof that verify passport
-	// need logic for check proof and accruing points for verification.
-
 	req, err := requests.NewVerifyPassport(r)
 	if err != nil {
 		Log(r).WithError(err).Debug("Bad request")
 		ape.RenderErr(w, problems.BadRequest(err)...)
 		return
 	}
 
-	log := Log(r).WithFields(map[string]any{
-		"nullifier":   req.Nullifier,
-		"hash":        req.Hash,
-		"shared_data": req.SharedData,
-	})
+	nullifier := req.Data.ID
 
-	balance, err := BalancesQ(r).FilterByPassportHash(req.Hash).Get()
-	if err != nil {
-		log.WithError(err).Error("Failed to get balance by Hash")
-		ape.RenderErr(w, problems.InternalError())
+	if !auth.Authenticates(UserClaims(r), auth.UserGrant(nullifier)) {
+		ape.RenderErr(w, problems.Unauthorized())
 		return
 	}
 
-	if balance != nil && balance.Nullifier != req.Nullifier {
-		log.Error("passport_hash already in use")
-		ape.RenderErr(w, problems.Conflict())
+	balance, err := BalancesQ(r).FilterByNullifier(nullifier).Get()
+	if err != nil {
+		Log(r).WithError(err).Error("Failed to get balance by nullifier")
+		ape.RenderErr(w, problems.InternalError())
 		return
 	}
 
 	if balance == nil {
-		balance, err = BalancesQ(r).FilterByNullifier(req.Nullifier).Get()
-		if err != nil {
-			log.WithError(err).Error("Failed to get balance by nullifier")
-			ape.RenderErr(w, problems.InternalError())
-			return
-		}
-	}
-
-	var reward *int64
-	evType := EventTypes(r).Get(evtypes.TypePassportScan, evtypes.FilterInactive)
-	if evType != nil {
-		var success bool
-		reward, success = EventTypes(r).CalculatePassportScanReward(req.SharedData...)
-		if !success {
-			log.Error("Failed to calculate PassportScanReward, incorrect fields")
-			ape.RenderErr(w, problems.NotFound())
-			return
-		}
+		Log(r).Debug("Balance absent")
+		ape.RenderErr(w, problems.NotFound())
+		return
 	}
 
-	if balance == nil {
-		log.Debug("Balance not found, creating new one")
-
-		err = createBalanceWithPassportTx(r, req, reward)
-
-		if err != nil {
-			log.WithError(err).Error("Failed to create balance with events")
-			ape.RenderErr(w, problems.InternalError())
-			return
-		}
-
-		w.WriteHeader(http.StatusNoContent)
+	if !balance.ReferredBy.Valid {
+		Log(r).Debug("Balance inactive")
+		ape.RenderErr(w, problems.BadRequest(errors.New("Balance inactive"))...)
```

Comment:
Don't pass anything except `BadRequester` or `validation.Errors` in `BadRequest`; don't capitalize error message